store.go

```diff
@@ -1786,6 +1799,65 @@ func (s *store) ApplyDiff(to string, diff archive.Reader) (int64, error) {
 	return -1, ErrLayerUnknown
 }
 
+func (s *store) layersByMappedDigest(m func(ROLayerStore, digest.Digest) ([]Layer, error), d digest.Digest) ([]Layer, error) {
+	var layers []Layer
+	rlstore, err := s.LayerStore()
+	if err != nil {
+		return nil, err
```

Comment:
You've a number of no-op's on errors scattered about. Would it be worthwhile (if possible) to at least add unique debug messages to them?